utils/route_diag/src/main.cpp

```diff
@@ -158,6 +158,9 @@ static void profile_source(const Netlist<>& net_list,
     vtr::ScopedStartFinishTimer timer("Profiling source");
     const auto& device_ctx = g_vpr_ctx.device();
     const auto& grid = device_ctx.grid;
+    // TODO: We assume if this function is called, the grid has a 2D structure
```

Comment:
This code is used for diagnostics / profiling of the router, so it's not on by default.  It assumes everything is on layer number 0, so it won't work yet for multi-layer FPGAs.